### PR TITLE
Explicitly close cursors

### PR DESCRIPTION
**Pull request by ueshin**
### Description

Explicitly closes cursors.

Sometimes we observe **Query has been timed out due to inactivity** error in the Query History UI, that is likely because the cursors are not closed.

### CODE REVIEW

**Review by zhengruifeng — APPROVED** [review]

File: dbt/include/databricks/macros/materializations/seed.sql [hunk 1]
```diff
@@ -29,7 +29,7 @@
           {%- endfor %}
       {% endset %}
 
-      {% do adapter.add_query(sql, bindings=bindings, abridge_sql_log=True) %}
+      {% do adapter.add_query(sql, bindings=bindings, abridge_sql_log=True, close_cursor=True) %}
```
Comment (allisonwang-db) [inline dbt/include/databricks/macros/materializations/seed.sql:32]: Why do we need to pass the `close_cursor` flag for seed?

File: dbt/adapters/databricks/connections.py [hunk 2]
```diff
@@ -210,21 +219,23 @@ def description(
...
+            self._cursor.close()
+            with warnings.catch_warnings():
+                warnings.simplefilter("always")
+                warnings.warn("The cursor was closed by destructor.")
+
 
 class DatabricksConnectionManager(SparkConnectionManager):
     TYPE: ClassVar[str] = "databricks"
 
-    DROP_JAVA_STACKTRACE_REGEX: ClassVar["re.Pattern[str]"] = re.compile(
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/connections.py:224]: This is deleted because we never used it?

File: dbt/adapters/databricks/connections.py [hunk 3]
```diff
@@ -141,46 +143,53 @@ def _connection_keys(self, *, with_aliases: bool = False) -> Tuple[str, ...]:
...
-
-    def cancel(self) -> None:
-        if self._cursor:
-            try:
-                self._cursor.cancel()
-            except DBSQLError as exc:
-                logger.debug("Exception while cancelling query: {}".format(exc))
-                _log_dbsql_errors(exc)
+    def cursor(self) -> "DatabricksSQLCursorWrapper":
+        return DatabricksSQLCursorWrapper(self._conn.cursor())
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/connections.py:155]: Should we avoid creating an instance of the wrapper each time we invoke the cursor?

File: dbt/adapters/databricks/connections.py (hunk 2 above)
Comment (ueshin) [reply dbt/adapters/databricks/connections.py:224]: Yes.

File: dbt/include/databricks/macros/materializations/seed.sql (hunk 1 above)
Comment (ueshin) [reply dbt/include/databricks/macros/materializations/seed.sql:32]: `adapter.add_query` doesn't close the cursor by default, then the cursor will remain opened.

File: dbt/adapters/databricks/connections.py (hunk 3 above)
Comment (ueshin) [reply dbt/adapters/databricks/connections.py:155]: How can we manage cursors otherwise?

Comment (allisonwang-db) [reply dbt/adapters/databricks/connections.py:155]: Nvm this method should return a new cursor.

File: dbt/include/databricks/macros/materializations/seed.sql (hunk 1 above)
Comment (allisonwang-db) [reply dbt/include/databricks/macros/materializations/seed.sql:32]: I see!

**Review by allisonwang-db — APPROVED** [review]
Looks good!

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**